File: Medium/11. Container With Most Water/Solution.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        left = 0
        right = len(height) - 1
        max_area = 0
        
        while left < right:
            # Genişlik * Kısa Olan Yükseklik
            width = right - left
            current_area = min(height[left], height[right]) * width
            max_area = max(max_area, current_area)
            
            # Kısa olan tarafı değiştir
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
                
        return max_area

File: Medium/11. Container With Most Water/Solution.cs
using System;

public class Solution {
    public int MaxArea(int[] height) {
        int left = 0;
        int right = height.Length - 1;
        int maxArea = 0;

        while (left < right) {
            int currentHeight = Math.Min(height[left], height[right]);
            int width = right - left;
            int currentArea = currentHeight * width;

            maxArea = Math.Max(maxArea, currentArea);

            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return maxArea;
    }
}